zebra-chain/src/sapling/tree.rs

```diff
@@ -287,6 +290,71 @@ pub struct NoteCommitmentTree {
     cached_root: std::sync::RwLock<Option<Root>>,
 }
 
+/// The block height of the final leaf in a level 16 subtree and
+/// the block height of the first leaf of the next subtree
+#[derive(Clone, Debug)]
+pub struct NoteCommitmentSubtree {
+    /// Index of this subtree
+    pub index: u16,
+    /// Start boundary of this subtree, the block height of its last leaf.
+    pub start: Option<block::Height>,
+    /// End boundary of this subtree, the block height of its last leaf.
+    pub end: block::Height,
+    /// Whether this subtree's end height is the start height of the next subtree.
+    pub begins_next_subtree: Option<bool>,
+    /// Root of this subtree.
+    pub subtree_root: Node,
+}
+
+impl NoteCommitmentSubtree {
+    /// Returns subtree bounds
+    pub fn bounds(&self) -> (NoteCommitmentSubtreeEnd, Option<NoteCommitmentSubtreeStart>) {
+        let NoteCommitmentSubtree {
+            index,
+            end,
+            begins_next_subtree,
+            subtree_root,
+            ..
+        } = *self;
+
+        (
+            NoteCommitmentSubtreeEnd {
+                index,
+                height: end,
+                root: subtree_root,
+            },
+            begins_next_subtree.and_then(|begins_next_subtree| {
+                begins_next_subtree.then_some(NoteCommitmentSubtreeStart {
+                    index: index + 1,
+                    height: end,
+                })
+            }),
+        )
+    }
+}
+
+/// The block height of the first leaf of a subtree
```

Comment:
I think `NoteCommitmentSubtreeStart` always refers to the next subtree.

Maybe something like this?

```suggestion
/// The index and block height of the next subtree.
```

Reply:
Applied in https://github.com/ZcashFoundation/zebra/pull/7334/commits/654b40fc16e2088e85b9d4f3dcbd453c123d0154.